### PR TITLE
refactor(lang): cover all China locale string types

### PR DESCRIPTION
**Pull request by JamesVu33**
update ZH langugage type detecting

### CODE REVIEW

File: lib/src/common/extensions/locale_x.dart [hunk 1]
```diff
@@ -48,12 +64,32 @@ extension StringLocale on String {
...
-          languageCode: components[0], scriptCode: components[1]);
+      return _getPizzaPlatformLanguage(components);
     } else if (components.isNotEmpty) {
       return Locale(components[0]);
     } else {
       return LocaleX.fallbackLocale;
     }
   }
+
+  Locale _getPizzaPlatformLanguage(List<String> components) {
```
Comment (duc-ios) [inline lib/src/common/extensions/locale_x.dart]: remove pizza word

File: lib/src/common/extensions/locale_x.dart [hunk 2]
```diff
@@ -34,6 +34,22 @@ extension LocaleX on Locale {
   String get fullLanguageCode =>
       [languageCode, scriptCode].where((element) => element != null).join('_');
 
+  bool isItemSelected(Locale compare) {
```
Comment (duc-ios) [inline lib/src/common/extensions/locale_x.dart]: shoud be renamed to `isEqualTo`

**Review by duc-ios — CHANGES_REQUESTED** [review]

File: lib/src/common/extensions/locale_x.dart (hunk 1 above)
Comment (JamesVu33) [reply lib/src/common/extensions/locale_x.dart]: updated

File: lib/src/common/extensions/locale_x.dart (hunk 2 above)
Comment (JamesVu33) [reply lib/src/common/extensions/locale_x.dart]: updated

**Pull request merged**